pkg/providers/imagefamily/bootstrap/aksbootstrap.go

```diff
@@ -191,63 +178,25 @@ var (
 			VnetCniPluginsUrl: vnetCNILinuxPluginsURL, // - [currently required, installCNI in provisioning scripts depends on CNI_PLUGINS_URL]
 			CniPluginsUrl:     cniPluginsURL,          // - [currently required, same]
 		},
-		ContainerdConfig: &nbcontractv1.ContainerdConfig{
-			ContainerdDownloadUrlBase: "", // -
-			ContainerdVersion:         "", // -
-			ContainerdPackageUrl:      "", // -
-		},
-		IsVhd: true, // s
+		ContainerdConfig: &nbcontractv1.ContainerdConfig{},
+		IsVhd:            true, // s
 		GpuConfig: &nbcontractv1.GPUConfig{
 			ConfigGpuDriver:    true,  // s
 			GpuDevicePlugin:    false, // -
 			GpuInstanceProfile: "",    // td
 		},
-		TeleportConfig: &nbcontractv1.TeleportConfig{
-			TeleportdPluginDownloadUrl: "",    // -
-			Status:                     false, // td
-		},
-		RuncConfig: &nbcontractv1.RuncConfig{
-			RuncVersion:    "", // -
-			RuncPackageUrl: "", // -
-		},
-		EnableSsh:              true,  // td
-		EnableHostsConfigAgent: false, // n
-		HttpProxyConfig: &nbcontractv1.HTTPProxyConfig{
-			HttpProxy:      "",           // cd
-			HttpsProxy:     "",           // cd
-			NoProxyEntries: []string{""}, // cd
-			ProxyTrustedCa: "",           // cd
-		},
-		CustomCaCerts:              []string{},                                  // cd
-		Ipv6DualStackEnabled:       false,                                       //s
-		OutboundCommand:            GetDefaultOutboundCommand(),                 // s
-		EnableUnattendedUpgrade:    false,                                       // cd
-		WorkloadRuntime:            nbcontractv1.WorkloadRuntime_WR_UNSPECIFIED, // s
-		AzurePrivateRegistryServer: "",                                          // cd
-		CustomSearchDomain: &nbcontractv1.CustomSearchDomain{
-			CustomSearchDomainName:          "", // cd
-			CustomSearchDomainRealmUser:     "", // cd
-			CustomSearchDomainRealmPassword: "", // cd
-		},
+		TeleportConfig:     &nbcontractv1.TeleportConfig{},
```

Comment:
Think we can remove TeleportConfig, RuncConfig, and CustomSearchDomain as well, since the getters are nil safe

Reply:
but these configs as well as ContainerdConfig and CustomCloudConfig from my other comments aren't used in `appyOptions` so shouldn't have to worry about them being nil there. The only place they are referenced is directly in the cse_cmd template in which we use getter's like so: `.TeleportConfig.GetStatus`, in this case the only getter that matters is for the fields in the config which are implemented like this:
```
func (x *TeleportConfig) GetStatus() bool {
	if x != nil {
		return x.Status
	}
	return false
}
```
Here, the getter checks if the object calling it is nil and if it is, it simply returns the default value, which is what we want. So, it should be okay to leave these configs nil, since none of the fields are specified and we just want to return the default values.

Reply:
You are right. I overlooked. I thought it was saying all configs but you didn't mean that. I removed all the unnecessary configs initialization and add unit test to cover it.